### ISSUE TITLE
wrong information

### ISSUE THREAD

**Issue by tugrul** [JamesMGreene/napi-async-promise-example#2]
> If this ever becomes UNTRUE, then you would need to utilize an AsyncWorker (or a similar concept) in order to ensure this Promise is fulfilled asynchronously.

Main process should not be blocked by the method if this idea is correct. But it blocks as you can see example below.

### Addon
```cpp

#include <thread>
#include <chrono>
#include <napi.h>

Napi::Promise GetNumberTen(const Napi::CallbackInfo& info) {

    Napi::Env env = info.Env();

    Napi::Promise::Deferred deferred = Napi::Promise::Deferred::New(env);

    uint8_t number = 0;

    std::thread runner([&number] {

        std::this_thread::sleep_for(std::chrono::seconds(2));

        number = 10;

    });

    runner.join();

    deferred.Resolve(Napi::Number::New(env, number));

    return deferred.Promise();
}

Napi::Object Init(Napi::Env env, Napi::Object exports) {

    exports.Set(Napi::String::New(env, "getNumberTen"), Napi::Function::New(env, GetNumberTen));

    return exports;
}

NODE_API_MODULE(addon, Init)

```

### Runner
```js
const { getNumberTen } = require('bindings')('addon.node');

console.time();

getNumberTen();

console.timeEnd();
```

### Output
```
default: 2.001s
```

**Comment by tugrul** [comment]
But your idea can be real if Promise api can be changed. It should have condition checker or something similar.

```cpp

Napi::Promise GetNumberTen(const Napi::CallbackInfo& info) {

    Napi::Env env = info.Env();

    Napi::Promise::Deferred deferred = Napi::Promise::Deferred::New(env);

    // imaginary condition checker method
    if (!deferred.Started()) {
        return deferred.Promise();
    }

    // second call (internally) of method will be non blocking 

    uint8_t number = 0;

    std::thread runner([&number] {

        std::this_thread::sleep_for(std::chrono::seconds(2));

        number = 10;

    });

    runner.join();

    deferred.Resolve(Napi::Number::New(env, number));

    return deferred.Promise();
}

```
